Std/Data/List/Basic.lean

```diff
@@ -823,6 +823,40 @@ inductive Forall₂ (R : α → β → Prop) : List α → List β → Prop
 
 attribute [simp] Forall₂.nil
 
+/--
+Decision procedure for `Forall₂ R l₁ l₂`
+-/
+def Forall₂.dec (R : α → β → Prop) [∀ a b, Decidable (R a b)] : List α → List β → Bool
```

Comment:
I think this would be more useful as a plain `Bool` function:
```lean
def all₂ (r : α → β → Bool) : List α → List β → Bool
```

Reply:
Do we need this function at all? I would guess you can just directly implement the Decidable instance without any intermediate declarations

Reply:
Kind of. It's harder to make the decidable instance tail recursive.

Reply:
yea, I changed it to guarantee that it's tail recursive